src/client/api/types.ts

```diff
@@ -19,13 +19,13 @@ export interface ILanguageServer extends Disposable {
 
 export const IPythonApiProvider = Symbol('IPythonApi');
 export interface IPythonApiProvider {
-    getApi(): Promise<PythonApi>;
+    getApi(): Promise<PythonApi | undefined>;
     setApi(api: PythonApi): void;
 }
 export const IPythonExtensionChecker = Symbol('IPythonExtensionChecker');
 export interface IPythonExtensionChecker {
     readonly isPythonExtensionInstalled: boolean;
-    showPythonExtensionInstallRequiredPrompt(): Promise<void>;
+    showPythonExtensionInstallRequiredPrompt(): Promise<boolean>;
```

Comment:
Can we use `InstallerResponse`, as the boolean still doesn't tell us much.
Does it mean it was installed or a message was displayed?